src/Utils/utils.ts

```diff
@@ -544,3 +544,25 @@ export const fahrenheitToCelsius = (fahrenheit: number) => {
 export const keysOf = <T extends object>(obj: T) => {
   return Object.keys(obj) as (keyof T)[];
 };
+
+/**
+ * Converts a string to snake_case format.
+ *
+ * @example
+ * toSnakeCase("Hello World!") // returns "hello_world"
+ * toSnakeCase("User-Name:") // returns "user_name"
+ *
+ * @param text - The input text to be converted
+ * @returns The text in snake_case format
+ * @throws {Error} If the input is null or undefined
+ */
+export const toSnakeCase = (text: string): string => {
+  if (text == null) {
+    throw new Error("Input text cannot be null or undefined");
+  }
```

Comment:
`text` is of type `string` and if things passed to it are type-safe, this would never be null in the first place.

```suggestion
 */
export const toSnakeCase = (text: string): string => {
```